src/requests/consumption.js

```diff
@@ -521,13 +521,34 @@ const createSolutions = async (resourceDetails, programDetails) => {
 				insertCertificateTemplate(found.certificate, found._id, programDetails._id)
 			})
 		}
+		const projectsCollection = mongoDb.collection(COLLECTIONS.TEMPLATES)
+		const createdSolutionsResponse = await Promise.all(
+			createdSolutions.map(async (solution) => {
+				const resultUpdateProjectTemplate = await projectsCollection.updateOne(
+					{
+						_id: solution.projectTemplateId,
+					},
+					{
+						$set: {
+							solutionId: solution._id,
+							solutionExternalId: solution.externalId,
+						},
+					}
+				)
 
-		return createdSolutions.map((solution) => {
-			return {
-				...solution,
-				rolloutId: solutionRolloutMap[solution.externalId],
-			}
-		})
+				// Validate the result of the template updation
+				if (!resultUpdateProjectTemplate) {
+					throw new Error(`Failed to update the template into the ${COLLECTIONS.SOLUTIONS} collection.`)
```

Comment:
@adithyadinesh0412  here you are updating the template with solution details right? Please change the message according to that